source/vistas/core/graphics/texture.py

```diff
@@ -14,16 +24,9 @@ def __init__(self, data=None, width=None, height=None, use_rgb=True):
         glBindTexture(GL_TEXTURE_2D, 0)
 
         if data is not None and width is not None and height is not None:
-            self.teximage2d(data, width, height, use_rgb)
+            glBindTexture(GL_TEXTURE_2D, self.texture)
+            glTexImage2D(GL_TEXTURE_2D, 0, src_format, width, height, 0, gl_format, GL_UNSIGNED_BYTE, data)
+            glBindTexture(GL_TEXTURE_2D, 0)
 
     def __del__(self):
         glDeleteTextures([self.texture])
-
-    def teximage2d(self, data, width, height, use_rgb=True):
```

Comment:
At first I was curious why this was being removed, but after a second pass I see why this isn't needed anymore. Whenever we have texture data that needs to be used in rendering, it's not all that common that we're going to be using the same texture data. Only case where this would be useful is in point selection and marking where we hit it. Even then, its likely faster to just allocate a new texture that the shader is pointing to, rather than making the round trip to get the data, edit it, and then resubmit it.

Reply:
It was mostly because it wasn't being used anywhere. I didn't consider the case of binding new data to an existing texture id, though I could see that being useful. I say we leave as-is for now and revisit support for re-using texture ids if the need arises later.

Reply:
Sounds good!